src/pages/account.js

```diff
@@ -1,28 +1,59 @@
-import useSWR from 'swr'
-import {CreateForm, EditForm} from '../components/channel-forms'
-import {supabase} from '../utils/supabase-client'
-
-// <Account> fetches the user's channel and shows either <CreateForm> or <EditForm>.
-
-const QUERY = () => supabase.from('channels').select(`*`).eq('user_id', supabase.auth.user().id)
-
-export default function Account({session}) {
-	const {data: channels, error, mutate} = useSWR(['userchannel', QUERY])
-
-	if (error) return <p>Error! {error.message}</p>
-	if (!channels) return <p>Loading...</p>
-	if (channels.length > 1)
-		return <p>You have more than one radio. That's weird and should not happen.</p>
-	if (!channels.length) return <CreateForm onCreate={(channel) => mutate({channel})}></CreateForm>
+import {useUserChannels} from '../hooks/use-user-channels.js'
+import {
+	createChannel,
+	updateChannel,
+	deleteChannel
+} from '../utils/crud/channel'
+import {
+	CreateForm,
+	UpdateForm,
+	DeleteForm
+} from '../components/channel-forms'
 
+export default function Account({dbSession}) {
+	const {session, database} = dbSession
+	const userId = session.user.id
```

Comment:
```suggestion
	const userId = database.auth.user().id
```

or am I missing something?

Reply:
why change from the `session` to `database`?